### PR TITLE
Prepare for cmcc 12

### PR DESCRIPTION
**Pull request by uwilken**
Update libs

tested with master and 2310

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -17,22 +17,22 @@ jobs:
     steps:
       - name: Checkout PR
         if: github.event_name == 'pull_request'
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.1.1
```
Comment (pitgrap) [inline .github/workflows/build.yml:20]: I would set @4 for rolling releases instead of pinning the version as we do it for all ther actions too.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -17,22 +17,22 @@ jobs:
...
         with:
           ref: ${{ github.event.pull_request.head.ref }}
       - name: Checkout
         if: github.event_name != 'pull_request'
         uses: actions/checkout@v3
       - name: Set up JDK 11
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v4.0.0
```
Comment (pitgrap) [inline .github/workflows/build.yml:27]: same here

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -41,12 +41,12 @@ jobs:
       - name: Setup PNPM
         uses: pnpm/action-setup@v2
         with:
-          version: 7.1.5
+          version: 8
 
       - name: Setup NodeJS
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v4
```
Comment (pitgrap) [inline .github/workflows/build.yml:47]: new version 4? we should update campaigns too!

**Review by pitgrap — APPROVED** [review]
Can't say anything tp pom changes. But if everything works, LGTM besides small remarks.

File: studio-client/apps/main/content-hub-adapter-rss/package.json [hunk 4]
```diff
@@ -5,18 +5,18 @@
   "license": "CoreMedia Open Source License",
   "private": true,
   "dependencies": {
-    "@coremedia/studio-client.core-icons": "^2110.2.0",
-    "@coremedia/studio-client.main.content-hub-editor-components": "^2110.2.0",
-    "@coremedia/studio-client.main.editor-components": "^2110.2.0",
-    "@jangaroo/runtime": "^1.0.0"
+    "@coremedia/studio-client.core-icons": "^2110.3.0",
```
Comment (rreich) [inline studio-client/apps/main/content-hub-adapter-rss/package.json:8]: why going to .3 here? Java dependencies stayed on 2, so I would also stay there for studio-client?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -41,12 +41,12 @@ jobs:
...
         with:
-          node-version: 16.x
+          node-version: 18
```
Comment (rreich) [inline .github/workflows/build.yml:49]: this should be the node version recommended for studio-client 2110.2 - whatever that is (but I suppose it is 16) - or is node always 100% backwards compatible?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -41,12 +41,12 @@ jobs:
       - name: Setup PNPM
         uses: pnpm/action-setup@v2
         with:
-          version: 7.1.5
+          version: 8
```
Comment (rreich) [inline .github/workflows/build.yml:44]: what is the pnpm version recommended for 2110.2?

File: studio-client/package.json [hunk 7]
```diff
@@ -3,17 +3,17 @@
   "version": "1.0.0",
   "private": true,
   "engines": {
-    "node": "16",
-    "pnpm": "^7.1.5"
+    "node": "18",
```
Comment (rreich) [inline studio-client/package.json:6]: same questions as in build.yml

File: studio-server/pom.xml [hunk 8]
```diff
@@ -104,6 +104,11 @@
       <artifactId>coremedia-common</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>com.coremedia.cms</groupId>
```
Comment (rreich) [inline studio-server/pom.xml:108]: why is this needed now?

File: .github/workflows/build.yml (hunk 1 above)
Comment (uwilken) [reply .github/workflows/build.yml:20]: 👍

File: .github/workflows/build.yml (hunk 2 above)
Comment (uwilken) [reply .github/workflows/build.yml:27]: 👍

File: studio-client/apps/main/content-hub-adapter-rss/package.json (hunk 4 above)
Comment (uwilken) [reply studio-client/apps/main/content-hub-adapter-rss/package.json:8]: yeah, but why not going to the freshest point release? that's what idea proposed

File: .github/workflows/build.yml (hunk 5 above)
Comment (uwilken) [reply .github/workflows/build.yml:49]: node 18 everywhere in the CM universe meanwhile.
I think we should aim to get this plugin working together with 2310 and master.
but yeah, it also should work together with older versions if possible.

@pitgrap maybe any thoughts?

File: studio-server/pom.xml (hunk 8 above)
Comment (uwilken) [reply studio-server/pom.xml:108]: mvn told me.
update of the dependency plugin version?
or 2110.2 vs 2110.3?

File: studio-client/apps/main/content-hub-adapter-rss/package.json (hunk 4 above)
Comment (pitgrap) [reply studio-client/apps/main/content-hub-adapter-rss/package.json:8]: I would update to .3 in java too.

File: .github/workflows/build.yml (hunk 5 above)
Comment (pitgrap) [reply .github/workflows/build.yml:49]: node16 is out of support. 

My recommendation:
- one major plugin release per major cmcc release. 
- for every cmcc major release always just support the latest LTS version:
  - CMCC 10 = 2107.X (node16 + java7?)
  - CMCC 11 = 2310.x (node18 + java11)
  - CMCC 12 = master (probably node20 + java17)

I don't know, how we could otherwise guarantee if it works. 
Most npm third-party libs strip support for old node releases.

If we don't care, we should at least test java and node versions via matrix in github action...

Comment (uwilken) [reply .github/workflows/build.yml:49]: @mkleine What do you think?

Comment (mkleine) [reply .github/workflows/build.yml:49]: I second Michael's recommendation. We should keep the main branch of the plugins up to date with respect of build and runtime dependencies. That's basically what we decided to do with the commerce adapters as well: Have a major version that is compatible with at least one CMCC major version. We try not to break the API, so we can tell customers that they can use any compatible version of their liking, be it the first one (built with old tooling, referencing the eldest compatible CM API) or a newer one (built with new tooling, referencing the current CM API).

File: .github/workflows/build.yml (hunk 1 above)
Comment (pitgrap) [reply .github/workflows/build.yml:20]: Did it get lost on rebasing? Still not "@4"

**Review by pitgrap — CHANGES_REQUESTED** [review]
still one version change in github action missing

File: .github/workflows/build.yml (hunk 1 above)
Comment (uwilken) [reply .github/workflows/build.yml]: now it is. gh actions do submit updated license info commits in this repo. change was not pushed.  now it should be there...

**Review by pitgrap — CHANGES_REQUESTED** [review]
Please change:
- Changelog for 2.1.0
- change the version to 2310 in README icon

File: studio-client/package.json (hunk 7 above)
Comment (uwilken) [reply studio-client/package.json:6]: node is used as a build tool. we will go with the newer version

File: .github/workflows/build.yml (hunk 5 above)
Comment (uwilken) [reply .github/workflows/build.yml:49]: after discussing this with (@fwienber was also involved) we will go back to the original cms version.

We want to stay backward compatible as long as possible.
All cms artifacts are provided by the runtime environment so we only need to bump the 3rd party libs which are packaged with the plugin. Since we do not use new/updated cms api in this pr.
As long as the plugin runs on newer versions we can avoid maintenance branches this way.

File: studio-server/pom.xml [hunk 9]
```diff
@@ -5,15 +5,15 @@
...
   <packaging>coremedia-plugin</packaging>
 
   <properties>
     <maven.compiler.release>11</maven.compiler.release>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <cm.studio-server.core.version>2110.2</cm.studio-server.core.version>
-    <cm.common.core.version>2110.2</cm.common.core.version>
+    <cm.studio-server.core.version>2310.2</cm.studio-server.core.version>
+    <cm.common.core.version>2310.2</cm.common.core.version>
```
Comment (mkleine) [inline studio-server/pom.xml:16]: didn't we agree that these should remain on 2110.2 ?

File: studio-server/pom.xml [hunk 10]
```diff
@@ -5,15 +5,15 @@
 
   <groupId>com.coremedia.labs.plugins</groupId>
   <artifactId>studio-server.content-hub-adapter-rss</artifactId>
-  <version>2.0.4-SNAPSHOT</version>
+  <version>2.1.0-SNAPSHOT</version>
```
Comment (mkleine) [inline studio-server/pom.xml:8]: didn't we agree that this version shouldn't be changed?

File: studio-client/apps/main/content-hub-adapter-rss/package.json [hunk 11]
```diff
@@ -1,22 +1,22 @@
...
   "private": true,
   "dependencies": {
-    "@coremedia/studio-client.core-icons": "^2110.2.0",
-    "@coremedia/studio-client.main.content-hub-editor-components": "^2110.2.0",
-    "@coremedia/studio-client.main.editor-components": "^2110.2.0",
-    "@jangaroo/runtime": "^1.0.0"
+    "@coremedia/studio-client.core-icons": "^2310.2.0",
+    "@coremedia/studio-client.main.content-hub-editor-components": "^2310.2.0",
+    "@coremedia/studio-client.main.editor-components": "^2310.2.0",
+    "@jangaroo/runtime": "^1.5.0"
```
Comment (mkleine) [inline studio-client/apps/main/content-hub-adapter-rss/package.json:11]: didn't we agree that these should remain on 2110.2 ?

File: studio-client/apps/main/content-hub-adapter-rss/package.json [hunk 12]
```diff
@@ -1,22 +1,22 @@
 {
   "name": "@coremedia-labs/studio-client.main.content-hub-adapter-rss",
-  "version": "2.0.4-SNAPSHOT",
+  "version": "2.1.0-SNAPSHOT",
```
Comment (mkleine) [inline studio-client/apps/main/content-hub-adapter-rss/package.json:3]: sure?

File: pom.xml [hunk 13]
```diff
@@ -5,7 +5,7 @@
 
   <groupId>com.coremedia.labs.plugins</groupId>
   <artifactId>content-hub-adapter-rss</artifactId>
-  <version>2.0.4-SNAPSHOT</version>
+  <version>2.1.0-SNAPSHOT</version>
```
Comment (mkleine) [inline pom.xml:8]: sure?

File: studio-client/apps/main/content-hub-adapter-rss/package.json (hunk 12 above)
Comment (uwilken) [reply studio-client/apps/main/content-hub-adapter-rss/package.json:3]: yes, we did. change not committed yet

**Review by mkleine — APPROVED** [review]
lgtm

File: studio-client/apps/main/content-hub-adapter-rss/package.json [hunk 14]
```diff
@@ -11,12 +11,12 @@
...
-    "@jangaroo/build": "^1.0.0",
-    "@jangaroo/core": "^1.0.0",
-    "@jangaroo/publish": "^1.0.0",
-    "@jangaroo/run": "^1.0.0",
-    "eslint": "7.27.0",
-    "rimraf": "^3.0.2"
+    "@jangaroo/build": "^1.5.0",
+    "@jangaroo/core": "^1.5.0",
+    "@jangaroo/publish": "^1.5.0",
+    "@jangaroo/run": "^1.5.0",
```
Comment (uwilken) [inline studio-client/apps/main/content-hub-adapter-rss/package.json:17]: @aholtkamp While discussing updating cm versions in plugins we came across node, pnpm and jangaroo versions. We want to stay compatible with old cms releases as long as possible and avoid cms release specific maintenance branches.

Quest now is, if the version updates in the package.json are reasonable for you? We left the jangaroo/runtime version unchanged to stay compatible with older studio installations. All the other dependencies should be used to build the plugin artefact internally.

File: .github/workflows/build.yml [hunk 15]
```diff
@@ -17,22 +17,22 @@ jobs:
     steps:
       - name: Checkout PR
         if: github.event_name == 'pull_request'
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.ref }}
       - name: Checkout
         if: github.event_name != 'pull_request'
         uses: actions/checkout@v3
```
Comment (pitgrap) [inline .github/workflows/build.yml:25]: Missed this action. Need to be v4 too. 
Why do we need 2 different checkouts here?

File: studio-client/apps/main/content-hub-adapter-rss/package.json (hunk 14 above)
Comment (aholtkamp) [reply studio-client/apps/main/content-hub-adapter-rss/package.json:17]: For runtime only the normal "dependencies" are relevant. So this change looks good to me if we want to avoid having a maintenance branch while still being compatible to older cms releases.

File: .github/workflows/build.yml (hunk 15 above)
Comment (uwilken) [reply .github/workflows/build.yml]: Version updated.
Cannot tell, why two different checkout...just bumping versions.
